GenericsHomework/Node.cs

```diff
@@ -0,0 +1,154 @@
+﻿using System.Collections;
+
+namespace GenericsHomework;
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1710:IdentifiersShouldHaveCorrectSuffix", Justification = "Benjamin is allowing this suppression.")]
+public class Node<T> : ICollection<T>
+{
+    private readonly T _value;
+    private Node<T> _next;
+    public Node(T value)
+    {
+        _value = value;
+        _next = this;
+    }
+
+    public T Value { get {  return _value; } }
+
+    public override string? ToString()
+    {
+        if (_value == null)
+        {
+            return null;
+        }
+        return _value.ToString();
```

Comment:
```suggestion
        return _value?.ToString();
```
This should be able to be simplified